Metricbeat system module updates for Windows

andrewkroh opened the pull request:
- Added system core metricset for Windows. Per core metrics were implemented
  in elastic/gosigar.
- Added logging of process/system details on Windows to aid in debugging (user,
  arch, cores, sid, privs).
- Fixes #2860 (PPID is zero on Windows).
- Fixes #1704 (Server 2003 - PID’s not recognized). The command line arguments
  for the process will not be reported on XP and 2003.
- Fixes #1897 (OpenProcess access denied on Windows). Added code to enable the
  SeDebugPrivilege when it is available.
- Fixes #2885 (Metricbeat diskio metricset doesn't work on Windows XP/2003).
- Enable fsstats by default in Metricbeat config file.

andrewkroh commented:
I would like to have elastic/gosigar#58 and #2282 (Metricbeat AppVeyor) merged first. Then I will update this PR with those fixes.

ruflin commented:
@andrewkroh Both PR's were merged.

ruflin commented on metricbeat/module/system/core/_meta/docs.asciidoc:
```diff
@@ -8,3 +8,4 @@ This metricset is available on:
 - FreeBSD
 - Linux
 - OpenBSD
+- Windows
```
👍

ruflin commented on metricbeat/etc/beat.short.yml:
```diff
@@ -1,58 +0,0 @@
-########################## Metricbeat Configuration ###########################
```
Is that change on purpose?

andrewkroh replied:
Yeah, this is on purpose, but I did want to ask you about it. I removed this file. Where is this file used?

ruflin commented on metricbeat/etc/beat.full.yml:
```diff
@@ -17,10 +17,10 @@ metricbeat.modules:
 
     # System Load stats
-    - load
+    #- load
```
Should we mention that we changed the defaults in the config file? Why is `load` disabled now by default?

tsg commented on metricbeat/module/system/_meta/config.full.yml:
```diff
@@ -16,7 +16,7 @@
 
     # File system summary stats
-    #- fsstat
+    - fsstat
```
Just making sure. We had this disabled before because it wasn't working on Windows and now it is?

andrewkroh replied:
It was working on Windows previously. So maybe it was commented out to reduce the amount of data Metricbeat sends? But the fsstats data is used on the Metricbeat overview dashboard which is why I think we should enabled it by default.

tsg commented on metricbeat/module/system/_meta/config.full.yml:
```diff
@@ -4,10 +4,10 @@
 
     # System Load stats
-    - load
+    #- load
```
I think we can leave this enabled. We have code to disable it automatically on WIndows: https://github.com/elastic/beats/blob/master/metricbeat/Makefile#L61

andrewkroh replied:
Yeah, I'll revert this. For some reason I thought we removed all of the OS specific `before-build` changes.

tsg commented on metricbeat/module/system/_meta/config.full.yml:
```diff
@@ -4,10 +4,10 @@
 
     # Per CPU core stats
-    #- core
+    - core
```
Also not sure about this one, I think we had it disabled by default in an effort to keep disk usage down?

andrewkroh replied:
I can revert this too. I was thinking that it was only commented out because of it not working on Windows.

andrewkroh commented:
I updated the PR to contain the gosigar fixes and rebased it so that it gets built on AppVeyor. I reverted the changes to `load` and `cores` in the config file. I have left `fsstats` enabled by default.
